node/src/types/status_feed.rs

```diff
@@ -35,9 +45,39 @@ static GET_STATUS_RESULT: Lazy<GetStatusResult> = Lazy::new(|| {
         chainspec_info: ChainspecInfo::doc_example().clone(),
         version: crate::VERSION_STRING.as_str(),
     };
-    GetStatusResult::from(status_feed)
+    GetStatusResult::new(status_feed, DOCS_EXAMPLE_PROTOCOL_VERSION.clone())
 });
 
+/// Summary information from the chainspec.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ChainspecInfo {
+    /// Name of the network.
+    name: String,
+    /// If `Some` then genesis or upgrade process returned a valid post state hash.
```

Comment:
What's the plan for emergency restarts: Will this also be `Some` then?

And for upgrades: I imagined we'd look into the latest block, and use its post-state hash, instead of the one found in the chainspec — but that works only for upgrades that don't migrate the state format; is the plan to always (automatically?) put the root hash into the chainspec somehow?

Reply:
> What's the plan for emergency restarts: Will this also be Some then?

Not the way the code is currently arranged in chainspec loader.

> And for upgrades: I imagined we'd look into the latest block, and use its post-state hash, instead of the one found in the chainspec — but that works only for upgrades that don't migrate the state format; is the plan to always (automatically?) put the root hash into the chainspec somehow?

There's no state hash in the chainspec, nor any plans to add one afaik.  This struct may be misnamed, since it does imply that the chainspec has one.  The root hash here is only held by the chainspec loader component in memory, and is the state hash returned by EE after performing the `commit_genesis()` or `upgrade()` work.

I think you're right in that it should never be `None` - we should be using the one from the highest block if we're not running `commit_genesis()` or `upgrade()`.  I'll update to handle that.